### PR TITLE
feat: add client-side error view

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds a dedicated error view for unexpected client-side errors.

## 🤷 Why

To improve the experience and tracking of unexpected client-side errors.

## 🛠️ How

- Adds custom `src/views/error-views/dev-dot-client`, and uses with `ErrorBoundary` in `_app.tsx`
- In `DevDotClient`, loads DataDog script tag
- In `DevDotClient` error view, tracks provided `Error` from `ErrorBoundary` with DataDog RUM
    - We're using the "async CDN" approach, so requires using the global `window.DD_RUM` rather than the library from `npm`. See [Collecting Browser Errors](https://docs.datadoghq.com/real_user_monitoring/browser/collecting_browser_errors/?tab=npm#collect-errors-manually) for details.

### Client Error View

client-error-view

### Error in DataDog

datadog-tracked-error

## 🧪 Testing

- [ ] Visit the [temporary preview link, `/test-error-view`][preview]
    - Ensure the client-side error view looks as expected
    - Try triggering an error, and check that it shows up in DataDog

[preview]: https://dev-portal-git-zsclient-error-view-hashicorp.vercel.app/test-error-view
[task]: https://app.asana.com/0/1202114367927919/1202405935832680/f

### CODE REVIEW

**Comment by zchsh** [comment]
@mhonnatti tagging you for early review here - re: how to track client-side errors (or more specifically, which Datadog method we should be calling rather than `userErrorPageAnalytics` as we do on other error views!)

File: src/pages/test-error-view.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useState } from 'react'
+
+function TestErrorView() {
```
Comment (zchsh) [inline src/pages/test-error-view.tsx:3]: 🚧 Will remove this `TestErrorView` before merging.

**Comment by zchsh** [comment]
@BRKalow @mhonnatti I tried to dive into learning a bit about DataDog here to get client-side errors tracked - let me know if this looks good to ya'll, and if you can also see the test errors showing up in DataDog 🙇‍♂️

**Review by BRKalow — APPROVED** [review]
👍  noice

**Review by mhonnatti — APPROVED** [review]

**Pull request merged**